modules/imaging_browser/php/imagingbrowserrowprovisioner.class.inc

```diff
@@ -146,14 +151,14 @@ class ImagingBrowserRowProvisioner extends \LORIS\Data\Provisioners\DBRowProvisi
         foreach ($case_desc as $key => $value) {
             $left_joins .= "
             LEFT JOIN (
-              SELECT files.SessionID, 
-                MIN(files_qcstatus.QCStatus+0) as $qc[$key] 
-              FROM files 
-              JOIN files_qcstatus USING (FileID) 
-              WHERE files.AcquisitionProtocolID= $key 
-                AND files_qcstatus.QCStatus IN (1, 2) 
-              GROUP BY files.SessionID) $pass[$key] 
-                ON ($pass[$key].SessionID=f.SessionID
+              SELECT files.SessionID,
+                MIN(files_qcstatus.QCStatus+0) as " . $DB->quote($qc[$key]) ." 
+              FROM files
+              JOIN files_qcstatus USING (FileID)
+              WHERE files.AcquisitionProtocolID= $key
+                AND files_qcstatus.QCStatus IN (1, 2)
+              GROUP BY files.SessionID) `$pass[$key]`
```

Comment:
Why isn't this using $DB->quote like the other places? Simply adding "`" isn't robust. For instance, it will still error out if the value has a "`" in it.

Reply:
I pushed a commit to call quote() at the top of the file for a cleaner look (L115 for the pass array). This makes all the elements inside the array quoted, let me know if this works. 

We still need the backticks however, removing them makes `$pass[$key]` get interpreted as a literal instead of an identifier, and spits out an error:

>Syntax error or access violation: 1064 You have an error in your SQL syntax; check the manual that corresponds to your MySQL server version for the right syntax to use near '''t2-defacedpass')'\n                ON (`'t2-defacedpass'`.SessionID=f.SessionID' at line 69

Reply:
sorry to have sent you down the wrong path.. `$DB->quote` is the wrong function, it's for quoting strings. It should be `$DB->escape` to escape table or column names.

```
   /**
     * Escapes a string properly for mysql and appends and prepends backticks.
     * Any backticks you append or prepend will be escaped.
     *
     * @param string $tableName The column or table name that needs to be escaped
     *
     * @return string surrounded by backticks and with special characters escaped.
     */
```

(luckily, it should be a simple change to change the places you called quote on a table/column name to escape..)